### ISSUE TITLE
Compilation error in generated code when Oracle OBJECT type contains index_of(varchar2) member procedure

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#9150]
When an object type like this is defined in Oracle:

```sql
CREATE OR REPLACE TYPE u_9150 AS OBJECT (
  dummy VARCHAR2(10),
  
  MEMBER FUNCTION index_of(arg VARCHAR2) RETURN NUMBER
)
/

CREATE OR REPLACE TYPE BODY u_9150 AS
  MEMBER FUNCTION index_of(arg VARCHAR2) RETURN NUMBER IS
  BEGIN
    RETURN 1;
  END index_of;
END;
/
```

Then a compilation error in generated code appears in jOOQ 3.12+ as the generated method returning a `BigDecimal` conflicts with `org.jooq.Record::indexOf`

**Comment by lukaseder** [comment]
So far, our logic in `GeneratorStrategyWrapper::disambiguateMethod` and then `::reservedColumns` only considered no-argument methods, because they cause much more trouble when accidentally overriding. It is more rare for a single-argument method like the above `indexOf()` method to produce a collision.

**Comment by lukaseder** [comment]
This could also happen with `STATIC` functions, in case of which the disambiguation would have to work slightly differently. Behind the scenes, a `MEMBER` function's first argument is `SELF`

**Comment by lukaseder** [comment]
This actually cannot happen to `STATIC` functions, as they always produce an `org.jooq.Configuration` as their first argument. There's no risk of any overriding errors when there's this type as a parameter type.

**Comment by lukaseder** [comment]
However, a similar issue can happen with `STATIC` functions: #9151

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.13.0. Backports scheduled for 3.12.2 (#9152) and 3.11.13 (#9153)
